File: backend/workers/common/cloud/CloudProviderFactory.py
from backend.workers.common.azure.BlobStorageService import BlobStorageService
from backend.workers.common.azure.ServiceBusService import ServiceBusService
from backend.workers.common.cloud.CloudProvider import CloudProvider
from backend.workers.common.cloud.CloudQueueService import CloudQueueService
from backend.workers.common.cloud.CloudStorageService import CloudStorageService

QUEUE_CONNECTION_STRING = "Endpoint=sb://revoicer.servicebus.windows.net/;" \
                          "SharedAccessKeyName=RootManageSharedAccessKey;" \
                          "SharedAccessKey=r62Vegf/HS2iKV6MJHaDprhNC6KSe/7jH+ASbAH7Sbc=;" \
                          "AccountName=revoicer"

STORAGE_CONNECTION_STRING = "DefaultEndpointsProtocol=https;" \
                            "AccountName=revoicer;" \
                            "AccountKey=0V+Dn/QnQ7tRSwaevHymXGtR/UgstoMITdhotUxGRtPDc5/wz+wj7QmQpHqWxP+N7eUI5LzHlwIp+AStdfhaKg==;" \
                            "EndpointSuffix=core.windows.net"

DEMUCS_INPUT_QUEUE_NAME = 'revoicer-demucs-input.fifo'
DEMUCS_OUTPUT_QUEUE_NAME = 'revoicer-demucs-output.fifo'

SVC_INPUT_QUEUE_NAME = 'revoicer-demucs-input.fifo'
SVC_OUTPUT_QUEUE_NAME = 'revoicer-demucs-output.fifo'

CONTAINER_NAME = "revoicer"


class CloudProviderFactory:
    def __init__(self, cloud_provider: CloudProvider):
        self.cloud_provider = cloud_provider

    def get_queue_service(self) -> CloudQueueService:
        match self.cloud_provider:
            case CloudProvider.Azure:
                return ServiceBusService(QUEUE_CONNECTION_STRING)
            case _:
                raise Exception("Provider not supported")

    def get_storage_service(self) -> CloudStorageService:
        match self.cloud_provider:
            case CloudProvider.Azure:
                return BlobStorageService(STORAGE_CONNECTION_STRING, CONTAINER_NAME)
            case _:
                raise Exception("Provider not supported")

File: backend/Ludikore.Revoicer/Ludikore.Revoicer.Services/Cloud/CloudProviderFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ludikore.Revoicer.Services.AWS;
using Ludikore.Revoicer.Services.Azure;

namespace Ludikore.Revoicer.Services.Cloud
{
    public class CloudProviderFactory
    {
        public CloudProvider Provider { get; }

        public CloudProviderFactory(CloudProvider provider)
        {
            Provider = provider;
        }

        public CloudQueueService GetQueueService()
        {
            switch (Provider)
            {
                case CloudProvider.AWS:
                    return new SqsService("http://localstack:4566");
                case CloudProvider.Azure:
                    return new ServiceBusService("Endpoint=sb://revoicer.servicebus.windows.net/;" +
                                                "SharedAccessKeyName=RootManageSharedAccessKey;" +
                                                "SharedAccessKey=r62Vegf/HS2iKV6MJHaDprhNC6KSe/7jH+ASbAH7Sbc=;" +
                                                "AccountName=revoicer");
                default:
                    throw new ArgumentOutOfRangeException($"There is no entry for this cloud provider: {Provider}");
            }
        }

        public CloudStorageService GetStorageService()
        {
            switch (Provider)
            {
                case CloudProvider.AWS:
                    return new S3Service("http://localstack:4566");
                case CloudProvider.Azure:
                    return new BlobStorageService(
                        connectionString: "DefaultEndpointsProtocol=https;" +
                                          "AccountName=revoicer;" +
                                          "AccountKey=0V+Dn/QnQ7tRSwaevHymXGtR/UgstoMITdhotUxGRtPDc5/wz+wj7QmQpHqWxP+N7eUI5LzHlwIp+AStdfhaKg==;" +
                                          "EndpointSuffix=core.windows.net",
                        sasAccountKey:
                        "0V+Dn/QnQ7tRSwaevHymXGtR/UgstoMITdhotUxGRtPDc5/wz+wj7QmQpHqWxP+N7eUI5LzHlwIp+AStdfhaKg==");
                default:
                    throw new ArgumentOutOfRangeException($"There is no entry for this cloud provider: {Provider}");
            }
        }
    }
}